pkg/services/contexthandler/contexthandler.go

```diff
@@ -451,12 +451,14 @@ func (h *ContextHandler) initContextWithRenderAuth(reqContext *models.ReqContext
 		return true
 	}
 
-	reqContext.IsSignedIn = true
-	reqContext.SignedInUser = &models.SignedInUser{
-		OrgId:   renderUser.OrgID,
-		UserId:  renderUser.UserID,
-		OrgRole: models.RoleType(renderUser.OrgRole),
+	query := models.GetSignedInUserQuery{UserId: renderUser.UserID, OrgId: renderUser.OrgID}
+	if err := h.SQLStore.GetSignedInUserWithCacheCtx(ctx, &query); err != nil {
```

Comment:
I remember one of the user queries had issues with retrieving service accounts - will this work if the `renderUser` points to a user thats actually a service account?

Reply:
Tested it and it works fine with service accounts.